### DISCUSSION TITLE
Home Assistant EVCC Addon - Modbus Proxy setup

### DISCUSSION THREAD

**Discussion by Cowboyschorle** [marq24/ha-evcc#45] [Q&A] [answered]
Dear all,

I have the following issue where I can`t find a solution for:
My Setup is as follows:
Sungrow Hybrid Inverter
BYD Battery
Go-e V3
Home Assistant on Raspberry Pi
EVCC Addon

In the beginning it worked like a charm but now I`m struggling with Errors in the Log.
I think this is due to Home Assistant fetching Data via Modbus from the Inverter and EVCC using the same Modbus.

I`m struggling to implement a Modbus Proxy. I`d like to fetch the data via the EVCC Plugin, as I want to control the Battery as well - from my knowledge this would not be possible, if the Proxy runs under Home Assistant.

ChatGPT was not helpful as well, so I`m seeking for help here.

Integration in Home Assistant is done via secrets.yaml:
sungrow_modbus_host_ip: 192.XXX.XXX.XX 
sungrow_modbus_port: 502 
sungrow_modbus_slave: 1 

evcc.yaml as follows:
  - name: my_grid
    type: template
    template: sungrow-hybrid
    usage: grid
    modbus: tcpip
    id: 1
    host: 192.XXX.XXX.XX  
    port: 502 
    timeout: 10s

  - name: my_pv
    type: template
    template: sungrow-hybrid
    usage: pv
    modbus: tcpip
    id: 1
    host: 192.XXX.XXX.XX  
    port: 502 
    timeout: 10s 
    
  - name: my_battery
    type: template
    template: sungrow-hybrid
    usage: battery
    modbus: tcpip
    id: 1
    host: 192.XXX.XXX.XX  
    port: 502 
    timeout: 10s 

Is there someone who has a similar Setup and can help me out? Is there any further information you need?

Cheers
Hannes

**Comment by marq24** [comment, marked as answer]
Hi Hannes ,

this sounds like a pure "evcc" question... I just can recommend to use the 'https://github.com/evcc-io/evcc/discussions'

**Reply by Cowboyschorle** [reply]
Thank you, posted it there :-) I`ll mark it as answered

**Comment by UDicke** [comment]
Hallo Hannes @Cowboyschorle,
bei uns läuft eine SAJ-H2 bzw. Ampere.Storage.Pro Batterie über Proxy in HA konfiguriert sowohl in HA als auch in evcc.
Die Konfiguration der Batterie habe ich im September von evcc.yaml auf "UI" umgestellt und sieht so aus...
Die IP ist die von HA und die PortNr. 503 im Proxy angelegt.
![grafik](https://github.com/user-attachments/assets/93fbeab6-3277-4827-9c18-5ee8243d2371)
Ich nutze dieses HA Add-on für den Proxy:
https://github.com/TCzerny/ha-modbusproxy
da es bis zu 3 Konfigurationsmöglichkeiten bietet und ich aktuell 2 (Mennekes Wallbox und Hausbatterie SAJ-H2) benutze.
Konfiguration des Modbus-Addon sieht etwa so aus...
Der erste Eintrag ist die Wallbox, der 2 Eintrag die Hausbatterie, der 3. Eintrag ist aus (disabled)
![grafik](https://github.com/user-attachments/assets/c2a02d3a-6f37-43b2-8c39-4d702c44f057)

Ich beutze allerdings die Daten vom Wechselrichter nur für die Batterie. Grid und PV vom Wechselrichter kann ich nicht gebrauchen, da ich zusätzliche Inverter im Hausnetz habe. Das kann der Wechselrichter der Batterie nicht wissen. Die Daten vom Grid und PV gehen bei uns über mqtt von HA an evcc.

Ich hoffe diese Info hilft Dir weiter. 😊

Gruß
Ulrich
